lib/wal/sources/Wal.hpp

```diff
@@ -0,0 +1,118 @@
+//
+// Created by Zoe Roux on 2021-05-14.
+//
+
+
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <typeinfo>
+#include <Exception/WalError.hpp>
+#include "Events/EventManager.hpp"
+#include "Renderer/Renderer.hpp"
+#include "Scene/SceneManager.hpp"
+#include "Entity/Entity.hpp"
+#include "System/System.hpp"
+
+namespace WAL
+{
+	//! @brief The main WAL class, it is used to setup and run the ECS.
+	class Wal
+	{
+	private:
+		//! @brief The scene manager that allow multiple scene to work together.
+		SceneManager _scenes;
+		//! @brief The event manager
+		EventManager _eventManager;
+		//! @brief The list of registered systems
+		std::vector<std::unique_ptr<System>> _systems = {};
+		//! @brief The renderer used to draw entities
+		std::unique_ptr<Renderer> _renderer;
```

Comment:
Renderer isn't just a system ?

Reply:
I hadded a onSelfUpdate() method called after every entity update so the renderer does not need to be a specific class now.